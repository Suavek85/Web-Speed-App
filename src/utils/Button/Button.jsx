import React from 'react'
import classNames from 'classnames/bind'
import styles from './Button.scss'
import * as signincontent from '../../constants/signinContent.js'

export default function Button(props) {

  const { 
    btnStyle, 
    handleOnClick, 
    isMobileCloseContent, 
    isDesktopCloseContent, 
    isFormContent } = props

  //STYLES
  const cx = classNames.bind(styles)
  const getClass = cx({
    block: true,
    blockRegular: btnStyle === 'regular',
    blockOval: btnStyle === 'oval',
  })

  return (
    <button 
      className={ getClass } 
      type={ isFormContent? 'submit' : 'button' } 
      onClick={ handleOnClick }
      value={ isFormContent? 'Submit form' : 'value' }>
      { isFormContent && signincontent.SUBMIT }
      { isMobileCloseContent && 'Close' }
      { isDesktopCloseContent && 'x' }
    </button>
  )
}