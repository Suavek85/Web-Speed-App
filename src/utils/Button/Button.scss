@import '../../scss/styles.scss';

$block: signin-btn;
$blockWarm: modifier($block, warm);
$blockCold: modifier($block, cold);

.#{$block} {
  width: 230px;
  height: 45px;
  padding: 6px 15px;
  margin-top: 10px;
  margin-bottom: 20px;
  font-size: em(14);
  font-weight: bold;
  color: #fff;
  cursor: pointer;
  border: 0;
  border-radius: 5px;
  outline: none;

  &--warm {
    background: #ee9ba3;
  }

  &--cold {
    background: #b8d1cb;
  }
}

:export {
  block: $block;
  blockWarm: $blockWarm;
  blockCold: $blockCold;
}
