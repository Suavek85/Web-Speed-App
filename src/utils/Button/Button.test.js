import React from 'react'
import { shallow } from 'enzyme'
import Button from './Button'

describe('Button snapshot', () => {

  it('should match the snapshot', () => {
    const wrapper = shallow(<Button />)
    expect(wrapper).toMatchSnapshot()
  })

})