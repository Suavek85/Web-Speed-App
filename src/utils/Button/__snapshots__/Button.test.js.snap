// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button snapshot should match the snapshot 1`] = `
<button
  className="block"
  type="button"
  value="value"
/>
`;
